File: 18-projects/python-sdk/projects.py
import os, requests
API = "https://cost-katana-backend.store/api"
KEY = os.getenv("COST_KATANA_API_KEY")

def create_project(name, description=None):
    res = requests.post(f"{API}/projects", 
        json={"name": name, "description": description},
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"✅ Project created: {res.json()['data']['id']}")
    return res.json()["data"]

def list_projects():
    res = requests.get(f"{API}/projects",
        headers={"Authorization": f"Bearer {KEY}"})
    res.raise_for_status()
    print(f"📋 Found {len(res.json()['data'])} projects")
    return res.json()["data"]

if __name__ == "__main__":
    create_project("Test Project", "Description")
    list_projects()

File: 18-projects/npm-package/projects.ts
import axios from 'axios';
const API = 'https://cost-katana-backend.store/api'\;
const KEY = process.env.COST_KATANA_API_KEY;

export async function createProject(name: string, description?: string) {
  const res = await axios.post(`${API}/projects`, {name, description}, 
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log('✅ Project created:', res.data.data.id);
  return res.data.data;
}

export async function listProjects() {
  const res = await axios.get(`${API}/projects`,
    {headers: {'Authorization': `Bearer ${KEY}`}});
  console.log(`📋 Found ${res.data.data.length} projects`);
  return res.data.data;
}

if (require.main === module) {
  (async () => {
    await createProject('Test Project', 'Description');
    await listProjects();
  })();
}
